sql/src/main/java/io/crate/analyze/UpdateAnalyzer.java

```diff
@@ -121,12 +125,29 @@ public AnalyzedUpdateStatement analyze(Update update, ParamTypeHints typeHints,
 
         Symbol normalizedQuery = normalizer.normalize(query, txnCtx);
 
-        SelectAnalysis selectAnalysis = SelectAnalyzer.analyzeSelectItems(update.returningClause(),
-                                                                          relCtx.sources(),
-                                                                          sourceExprAnalyzer,
-                                                                          exprCtx);
+        List<Symbol> outputSymbol = null;
+        Fields outputFields = null;
+
+        if (!update.returningClause().isEmpty()) {
+            SelectAnalysis selectAnalysis = SelectAnalyzer.analyzeSelectItems(update.returningClause(),
+                                                                              relCtx.sources(),
+                                                                              sourceExprAnalyzer,
+                                                                              exprCtx);
+
+            outputSymbol = Lists2.map(selectAnalysis.outputSymbols(), x -> normalizer.normalize(x, txnCtx));
+
+            outputFields = new Fields(selectAnalysis.outputNames().size());
+            Iterator<Symbol> outputsIterator = outputSymbol.iterator();
+            for (ColumnIdent path : selectAnalysis.outputNames()) {
+                outputFields.add(new Field(relation, path, outputsIterator.next()));
+            }
```

Comment:
Usually we build the fields inside the statements constructor, and the relation that they point to is `this`. See for example the `QueriedSelectRelation`.

I think we should also do that in the update case, as it makes it harder to build a "invalid" `AnalyzedUpdateStatement` instance.